boot_shutdown_manager/src/boot_shutdown_manager_core.cpp

```diff
@@ -109,28 +115,36 @@ BootShutdownManager::BootShutdownManager()
     this->get_parameter_or(
       param_prefix + "execute", execute_service_name,
       fmt::format("/api/{}/execute_shutdown", ecu_name));
+    this->get_parameter_or(
+      param_prefix + "service_address", service_address, std::string("127.0.0.1"));
+    this->get_parameter_or(
+      param_prefix + "prepare_shutdown_port", prepare_shutdown_port,
+      static_cast<unsigned short>(10001));
+    this->get_parameter_or(
+      param_prefix + "execute_shutdown_port", execute_shutdown_port,
+      static_cast<unsigned short>(10002));
 
     auto client = std::make_shared<EcuClient>();
     {
       client->skip_shutdown = skip_shutdown;
-      client->cli_execute = create_client<ExecuteShutdown>(
-        execute_service_name, rmw_qos_profile_services_default, callback_group_);
-      client->cli_prepare =
-        create_client<PrepareShutdown>(prepare_service_name, rmw_qos_profile_services_default);
-      client->sub_ecu_state = create_subscription<EcuState>(
-        state_topic_name, rclcpp::QoS{1},
-        [this, client](const EcuState::SharedPtr msg) { client->ecu_state = msg; });
-      client->ecu_state = std::make_shared<EcuState>();
-      client->ecu_state->state = EcuState::UNKNOWN;
-      client->ecu_state->name = ecu_name;
+      client->cli_execute = ServiceClient<ExecuteShutdownService>::create_client(
+        execute_service_name, io_context_, service_address, execute_shutdown_port,
+        service_timeout_);
+      client->cli_prepare = ServiceClient<PrepareShutdownService>::create_client(
+        prepare_service_name, io_context_, service_address, prepare_shutdown_port,
+        service_timeout_);
+      client->sub_ecu_state = TopicSubscriber<EcuStateMessage>::create_subscription(
+        state_topic_name, io_context_, topic_port_,
+        [this, client](const EcuStateMessage & message) { client->ecu_state = message; });
+      client->ecu_state.state = EcuStateType::UNKNOWN;
+      client->ecu_state.name = ecu_name;
     }
-    if(client->primary){
+    if (client->primary) {
```

Comment:
Is it necessary that `client->primary=primary`？

Reply:
The original code was missing. Thanks!
Fixed in https://github.com/tier4/boot_shutdown_tools/pull/30/commits/620c57db3830e43d1657370e9822fc89961c438e